cmd/crates/soroban-spec-typescript/src/project_template/src/assembled-tx.ts

```diff
@@ -0,0 +1,706 @@
+      )
+    }
+  }
+
+  get isReadCall(): boolean {
+    const authsCount = this.simulationData.result.auth.length;
+    const writeLength = this.simulationData.transactionData.resources().footprint().readWrite().length
+    return (authsCount === 0) && (writeLength === 0);
+  }
+
+  hasRealInvoker = async (): Promise<boolean> => {
+    const account = await this.getAccount()
+    return account.accountId() !== NULL_ACCOUNT
+  }
+}
+
+/**
+ * A transaction that has been sent to the Soroban network. This happens in two steps:
+ *
+ * 1. `sendTransaction`: initial submission of the transaction to the network.
+ *    This step can run into problems, and will be retried with exponential
+ *    backoff if it does. See all attempts in `sendTransactionResponseAll` and the
+ *    most recent attempt in `sendTransactionResponse`.
+ * 2. `getTransaction`: once the transaction has been submitted to the network
+ *    successfully, you need to wait for it to finalize to get the results of the
+ *    transaction. This step can also run into problems, and will be retried with
+ *    exponential backoff if it does. See all attempts in
+ *    `getTransactionResponseAll` and the most recent attempt in
+ *    `getTransactionResponse`.
+ */
+class SentTransaction<T> {
+  public server: Server
+  public signed: Tx
+  public sendTransactionResponse?: SendTx
+  public sendTransactionResponseAll?: SendTx[]
+  public getTransactionResponse?: GetTx
+  public getTransactionResponseAll?: GetTx[]
+
+  constructor(public options: AssembledTransactionOptions<T>, public assembled: AssembledTransaction<T>) {
+    this.server = new Server(this.options.rpcUrl, {
+      allowHttp: this.options.rpcUrl.startsWith("http://"),
+    });
+    this.assembled = assembled
+  }
+
+  static init = async <T>(
+    options: AssembledTransactionOptions<T>,
+    assembled: AssembledTransaction<T>,
+    secondsToWait: number = 10
+  ): Promise<SentTransaction<T>> => {
+    const tx = new SentTransaction(options, assembled)
+    return await tx.send(secondsToWait)
+  }
+
+  private send = async (secondsToWait: number = 10): Promise<this> => {
+    const wallet = await getWallet(this.options.wallet)
+
+    this.sendTransactionResponseAll = await withExponentialBackoff(
+      async (previousFailure) => {
+        if (previousFailure) {
+          // Increment transaction sequence number and resimulate before trying again
+
+          // Soroban transaction can only have 1 operation
+          const op = this.assembled.raw.operations[0] as Operation.InvokeHostFunction;
+
+          this.assembled.raw = new TransactionBuilder(await this.assembled.getAccount(), {
+            fee: this.assembled.raw.fee,
+            networkPassphrase: this.options.networkPassphrase,
+          })
+            .setTimeout(TimeoutInfinite)
+            .addOperation(
+              Operation.invokeHostFunction({ ...op, auth: op.auth ?? [] }),
+            )
+            .build()
```

Comment:
I know you mentioned this briefly on Discord, but can you elaborate why

```typescript
this.assembled.raw = TransactionBuilder.cloneFrom(this.assembled.raw);
```

does not work for you? It should be functionally identical.

Reply:
It does not increment the transaction sequence number

Reply:
hm, then you could call 

```
let account = await this.assembled.getAccount(); 
account.incrementSequenceNumber(); 
this.assembled.raw = TransactionBuilder.cloneFrom(...);
```

Reply:
I just double-checked this. It does not work. When I run all the tests in parallel, I get errors on most runs if I use the `cloneFrom` approach. Rebuilding it from the operations and re-simulating passes every time.